SpatialGDK/Source/SpatialGDKServices/Private/LocalDeploymentManager.cpp

```diff
@@ -147,6 +156,116 @@ void FLocalDeploymentManager::RefreshServiceStatus()
 	});
 }
 
+bool FLocalDeploymentManager::CheckIfPortIsBound(int32 Port)
+{
+	ISocketSubsystem* SocketSubsystem = ISocketSubsystem::Get(PLATFORM_SOCKETSUBSYSTEM);
+	bool bCanBindToPort = false;
+
+	// Set our broadcast address.
+	TSharedRef<FInternetAddr> BroadcastAddr = SocketSubsystem->CreateInternetAddr();
+	BroadcastAddr->SetBroadcastAddress();
+	BroadcastAddr->SetPort(Port);
+	
+	// Now the listen address.
+	TSharedRef<FInternetAddr> ListenAddr = SocketSubsystem->GetLocalBindAddr(*GLog);
+	ListenAddr->SetPort(Port);
+	const FString SocketName = TEXT("Runtime Port Test");
+	// Now create and set up our sockets.
+	FSocket* ListenSocket = SocketSubsystem->CreateSocket(NAME_Stream, SocketName, false /* bForceUDP */);
+	if (ListenSocket != nullptr)
+	{
+		ListenSocket->SetReuseAddr();
+		ListenSocket->SetNonBlocking();
+		ListenSocket->SetRecvErr();
+	
+		// Bind to our listen port.
+		if (ListenSocket->Bind(*ListenAddr))
+		{
+			bCanBindToPort = ListenSocket->Listen(0 /* MaxBacklog*/ );
+			ListenSocket->Close();
+		}
+		else
+		{
+			UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Failed to bind listen socket to addr (%s) for %s, the port is likely in use"), *ListenAddr->ToString(true), *SocketName);
+		}
+	}
+	else
+	{
+		UE_LOG(LogSpatialDeploymentManager, Verbose, TEXT("Failed to create listen socket for %s, the port is likely in use"), *SocketName);
+	}
+
+	// Either we couldn't create the socket or couldn't listen on it so the port is probably bound.
+	return !bCanBindToPort;
+}
+
+bool FLocalDeploymentManager::TryUnbindPort(int32 Port)
```

Comment:
`TryUnbindPort` sounds rather friendly for a method which runs taskkill :D 
Maybe we should change this to "KillProcessBlockingPort" to be more descriptive (don't want someone using this in the future and fires start)